internal/presto/domain.go

```diff
@@ -32,25 +32,44 @@ func NewDomain(hashKey, sortKey string, filters ...string) (*PrestoThriftTupleDo
 		}
 
 		tokens := ex.Tokens()
-		if len(tokens) != 3 {
+		if len(tokens) != 3 && len(tokens) != 7 {
```

Comment:
We can replace this with a switch on len(tokens) and error out in the default case.

Reply:
done.